docs/source/tutorials/notifications.rst

```diff
@@ -0,0 +1,217 @@
+Notifications
+=============
+
+If you read through :ref:`ref-chatops` section, you are familiar with notifications.
+Even without chatops, notifications can be used to post messages to external systems
+like chat clients, send emails etc. Notifications require an action that is registered with
+st2 (For example, ``slack`` pack contains a
+`post_message <https://github.com/StackStorm/st2contrib/blob/master/packs/slack/actions/post_message.yaml>`_ action.)
+and a notification rule to go with it. Notifications are implemented as triggers and rules and actions.
+A speical ``core.st2.notifytrigger`` is emitted by the system on completion of every action
+and a rule to match the trigger to a notify action results in notifications being sent out.
+
+How to setup a notification for a simple action?
+------------------------------------------------
+
+This is the easiest case. You can do this by specifying a ``notify`` section in the YAML meta
+while registering the action. For example:
+
+::
+
+    ---
+    description: Action that executes an arbitrary Linux command on the localhost.
+    enabled: true
+    entry_point: ''
+    name: local-notify
+    notify:
+      on-complete:
+        routes:
+        - slack
+        message: '"@channel: Action succeeded."'
+    parameters:
+      cmd:
+        description: Arbitrary Linux command to be executed on the remote host(s).
+        required: true
+        type: string
+      sudo:
+        immutable: true
+    runner_type: "local-shell-cmd"
+
+Above is the same action as a ``local-shell-cmd`` action but with notify. As you can see, there
+is a notify section with ``on-complete`` section. You can also specify `on-success`
+and ``on-failure`` sections with different messages. These subsections are all optional but at
+least one is required for any meaningful notification. For sake of clarity, an ``on-success`` case
+is presented below.
+
+
+::
+
+   notify:
+      on-complete:
+        routes:
+        - slack
+        message: '"@channel: Action succeeded."'
+      on-success:
+        routes:
+        - slack
+        message: '"@channel: Woohoo!"'
+
+The message doesn't support jinja templating yet. This support will be added in the future.
+Also, when the notification triggers are sent out, the message supplied along with a ``data``
+field containing the results of the execution are sent out. The rule can use these two fields -
+``message`` and ``data`` - and send it out as part of the action.
+
+How to write a rule for notification?
+-------------------------------------
+
+The rule to tie a st2 registered ``notify`` action resembles the notify rule you are familiar
+with when you setup chatops. An example is below:
+
+::
+
+    ---
+    name: "sample.notify_slack"
+    pack: "examples"
+    description: "Sample rule firing on action completion."
+    enabled: true
+
+    trigger:
+      type: "core.st2.generic.notifytrigger"
+      parameters: {}
+    criteria:
+      trigger.channel:
```

Comment:
So the trigger field is still called `channel` and not `route` or is this a typo?